src/app/services/http-service.ts

```diff
@@ -54,7 +54,7 @@ export class HttpService extends Http {
       console.log('retryWhen callback');
       let count = 0;
       return attempts.flatMap(error => {
-        if (error.status == 0) { // Server offline :: keep trying
+        if (error.status == 0 && count < 3) { // Server offline :: keep trying
```

Comment:
error.status == 0 comes when internet is gone. We need to keep retrying when this happens. Not just thrice.